docs/developer/design/authentication.rst

```diff
@@ -0,0 +1,154 @@
+Authentication
+==============
+
+Todo
+----
+
+- Provide uml diagrams for Dex + MetalK8s authentication flow
```

Comment:
Not sure it's really required, we can link to the relevant parts of Dex / K8s docs regarding OIDC flow.